docs/kubernetes/backup.md

```diff
@@ -0,0 +1,174 @@
+# Filestore Backups User Guide (Beta)
+
+>**Attention:** VolumeSnapshot is a Beta feature enabled by default in
+Kubernetes 1.17+.
+
+>**Attention:** VolumeSnapshot is only available in the driver version "master".
+
+### Backup Example
+A Filestore backup is a copy of a file share that includes all file data and metadata of the file share from the point in time when the backup is created. It works for Basic HDD and Basic SSD tier instances. Once a backup of a file share is created, the original file share can be modified or deleted without affecting the backup. A file share can be completely restored from a backup as a new Filestore instance or onto an existing file share. For more details refer to this [documentation](https://cloud.google.com/filestore/docs/backups)
+
+[CSI Snapshot](https://github.com/container-storage-interface/spec/blob/master/spec.md#createsnapshot) feature is leveraged to create Filestore Backups. By specifying a `type: backup` field in the VolumeSnapshotClass parameters, filestore CSI driver understands how to initiate a backup for a Filestore instance backed by the Persistent Volume.
+
+1. Create `StorageClass`
+
+    If you haven't created a `StorageClass` yet, create one first:
+
+    ```console
+    kubectl apply -f ./examples/kubernetes/backups-restore/sc.yaml
+    ```
+
+2. Create default `VolumeSnapshotClass`
+
+    ```console
+    kubectl create -f ./examples/kubernetes/backups-restore/backup-volumesnapshotclass.yaml
+    ```
+
+3. Create source PVC and Pod
+
+    ```console
+    kubectl create -f ./examples/kubernetes/backups-restore/source-pod-pvc.yaml
+    ```
+
+4. Generate sample data
+
+    Create a sample pod with the source PVC. The source PVC is mounted into `/demo/data` directory of this pod. This pod will create a file `sample-file.txt` in `/demo/data` directory.
+
+    Check if the file has been created successfully:
+
+    ```console
+    kubectl exec source-pod -- ls /demo/data/
+    ```
+
+    The output should be:
+
+    ```
+    lost+found
+    sample-file.txt
+    ```
+
+5. Create a `VolumeSnapshot` of the source PVC (This internally generates Filestore instance backup)
+
+    ```console
+    kubectl create -f ./examples/kubernetes/backups-restore/backup.yaml
+    ```
+
+6. Verify that `VolumeSnapshot` has been created and it is ready to use:
+
+    ```console
+    kubectl get volumesnapshot backup-source-pvc -o yaml
+    ```
+
+    The output is similar to this:
+
+    ```yaml
+    {
```

Comment:
I'm curious where you got this formatted output - we have similar output but my -o yaml output looks slightly different than this, eg.:
```
apiVersion: snapshot.storage.k8s.io/v1beta1
kind: VolumeSnapshot
metadata:
  creationTimestamp: "2020-11-11T21:41:10Z"
  finalizers:
  - snapshot.storage.kubernetes.io/volumesnapshot-as-source-protection
  - snapshot.storage.kubernetes.io/volumesnapshot-bound-protection
  generation: 1
  name: backup-source-pvc
  namespace: default
  resourceVersion: "29851"
  selfLink: /apis/snapshot.storage.k8s.io/v1beta1/namespaces/default/volumesnapshots/backup-source-pvc
  uid: ea963e5e-34fc-46d4-8420-2f944e471927
spec:
  source:
    persistentVolumeClaimName: source-pvc
  volumeSnapshotClassName: csi-gcp-filestore-backup-snap-class
status:
  boundVolumeSnapshotContentName: snapcontent-ea963e5e-34fc-46d4-8420-2f944e471927
  creationTime: "2020-11-11T22:11:29Z"
  readyToUse: true
  restoreSize: 1Ti
```

Reply:
That looks like -o json and not -o yaml?

I'd also elide away most of the fields except the spec and the status.

Reply:
the output was -o json. 
changed it to -o yaml and truncated some of the fields